src/bench/crypto_hash.cpp

```diff
@@ -192,10 +192,14 @@ static void SHA512(benchmark::Bench& bench)
 
 static void SipHash_32b(benchmark::Bench& bench)
 {
-    uint256 x;
-    uint64_t k1 = 0;
+    FastRandomContext rng(true);
+    auto k0 = rng.rand64(), k1 = rng.rand64();
+    auto val = rng.rand256();
     bench.run([&] {
-        *((uint64_t*)x.begin()) = SipHashUint256(0, ++k1, x);
+        ankerl::nanobench::doNotOptimizeAway(SipHashUint256(k0, k1, val));
+        ++k0;
+        ++k1;
+        ++(*val.begin());
```

Comment:
I couldn't reproduce the warning locally, but changed it to modify every value in `val.data()` instead to make sure none of them can be optimized away - would you be so kind and check if this solves the undefined behavior?

Reply:
i'm very confused about this one; `begin()` returns a `unsigned char*`, and unsigned values are definitely allowed to overflow within defined behavior. Also, `m_data` is an array so by definition [0] will not be out of bounds.

Reply:
a workaround may be to do `val.data()[0] = (val.data()[0] + 1) & 0xff;`  to make the wrap-around explicit, the C++ compiler will almost certainly just optimize it away

Reply:
Would you prefer I '& 0xFF' the current implementation?

Reply:
dunno, i would prefer `&0xff` to adding a supression, it more explicitly defines what is expected

Reply:
Thanks for the hints, I've replaced the increment of the bytes with a xor, this should keep the values in check. The overflow of `i` should be well-defined in this case, please let me know if you think any other change is needed since I couldn't reproduce the sanitizer warnings.
I have checked to make sure this still helps with the benchmarking of https://github.com/bitcoin/bitcoin/pull/30442.
Done in https://github.com/bitcoin/bitcoin/compare/c88b1bde7500ace70a694f36a82521f92221936c..100cded580cfe9d69cc30866c29697d9658b4ce3
(also change the brace init since the spirit of @hodlinator was haunting me)

Reply:
So are you ok with the current version?

Reply:
Thanks a lot @maflcko, that reproducer was super useful!
Added a `& 0xFF` (or do we prefer `0xff`? Found about the same number of occurrences in the code) and bumped the uint8_t to get rid of the overflow warning, see: https://github.com/bitcoin/bitcoin/compare/100cded580cfe9d69cc30866c29697d9658b4ce3..42066f45ff5d48e78a317eda63c035809bd657c6